src/libraries/System.Text.Json/tests/Serialization/PropertyVisibilityTests.cs

```diff
@@ -2161,20 +2165,193 @@ private struct SmallStructWithValueAndReferenceTypes
+            string expectedJson = @"{
+""MyPointClass2"":{},
+""MyString1"":""Default"",
+""MyInt2"":0,
+""MyPointStruct2"":{""X"":1,""Y"":2},
+""MyInt1"":1,
+""MyBool2"":true,
+""MyPointStruct1"":{""X"":0,""Y"":0}
+}";
+            JsonTestHelper.AssertJsonEqual(expectedJson, JsonSerializer.Serialize(obj, options));
+        }
+
+        public class ClassWithThingsToIgnore
+        {
+            public string MyString1 { get; set; }
+            [JsonInclude]
+            public string MyString2;
+            [JsonInclude]
+            public int MyInt1;
+            public int MyInt2 { get; set; }
+            public bool? MyBool1 { get; set; }
+            [JsonInclude]
+            public bool? MyBool2;
+            [JsonInclude]
+            public PointClass MyPointClass1;
+            public PointClass MyPointClass2 { get; set; }
+            [JsonInclude]
+            public Point_2D_Struct_WithAttribute MyPointStruct1;
+            public Point_2D_Struct_WithAttribute MyPointStruct2 { get; set; }
+        }
+
+        [Fact]
+        public static void Ignore_WhenWriting_Default_PerProperty()
+        {
+            string json = @"{
+""MyPointClass2"":{},
+""MyString1"":""Default"",
+""MyBool1"":null,
+""MyInt2"":0,
+""MyPointStruct2"":{""X"":1,""Y"":2},
+""MyInt1"":1,
+""MyString2"":null,
+""MyPointClass1"":null,
+""MyBool2"":true,
+""MyPointStruct1"":{""X"":0,""Y"":0}
+}";
+
+            // All members should correspond to JSON contents, as ignore doesn't apply to deserialization.
+            ClassWithThingsToIgnore_PerProperty obj = JsonSerializer.Deserialize<ClassWithThingsToIgnore_PerProperty>(json);
+            Assert.NotNull(obj.MyPointClass2);
+            Assert.Equal("Default", obj.MyString1);
+            Assert.Null(obj.MyBool1);
+            Assert.Equal(0, obj.MyInt2);
+            Assert.Equal(1, obj.MyPointStruct2.X);
+            Assert.Equal(2, obj.MyPointStruct2.Y);
+            Assert.Equal(1, obj.MyInt1);
+            Assert.Null(obj.MyString2);
+            Assert.Null(obj.MyPointClass1);
+            Assert.True(obj.MyBool2);
+            Assert.Equal(0, obj.MyPointStruct1.X);
+            Assert.Equal(0, obj.MyPointStruct1.Y);
+
+            // Ignore null as appropriate during serialization.
+            string expectedJson = @"{
+""MyPointClass2"":{},
+""MyString1"":""Default"",
+""MyInt2"":0,
+""MyPointStruct2"":{""X"":1,""Y"":2},
+""MyInt1"":1,
+""MyBool2"":true,
+""MyPointStruct1"":{""X"":0,""Y"":0}
+}";
+            JsonTestHelper.AssertJsonEqual(expectedJson, JsonSerializer.Serialize(obj));
+        }
+
+        public class ClassWithThingsToIgnore_PerProperty
+        {
+            [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+            public string MyString1 { get; set; }
+            [JsonInclude]
+            [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+            public string MyString2;
+            [JsonInclude]
+            public int MyInt1;
```

Comment:
This is hard to understand tests since I can't tell from the class name or the member name what is ignored or not. Suggest naming members like "MyString2_IgnoredWhenWritingNull"

Reply:
I've added `_IgnoredWhenWritingNull` where applicable. I also used `JsonSerializerOptions.IncludeFields` to reduce the `[JsonInclude]` noise per-field.